dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -1167,22 +1170,44 @@ private IHTMLPage loadPageByIdentifier ( String ident, boolean live, User user,
 
         final List<MultiTree> trees = APILocator.getMultiTreeAPI().getMultiTreesByChild(id.getId());
         for (final MultiTree tree : trees) {
-            final IHTMLPage page = APILocator.getHTMLPageAssetAPI()
-                    .findByIdLanguageFallback(tree.getParent1(), contentlet.getLanguageId(), false,
-                            APILocator.getUserAPI().getSystemUser(), false);
-            final Container container = APILocator.getContainerAPI()
-                    .getWorkingContainerById(tree.getParent2(),
-                            APILocator.getUserAPI().getSystemUser(), false);
-            if (InodeUtils.isSet(page.getInode()) && InodeUtils.isSet(container.getInode())) {
-                final Map<String, Object> map = new HashMap<String, Object>();
-                map.put("page", page);
-                map.put("container", container);
-                results.add(map);
+
+            try {
+                final IHTMLPage page = APILocator.getHTMLPageAssetAPI()
+                        .findByIdLanguageFallback(tree.getHtmlPage(), contentlet.getLanguageId(),
+                                false,
+                                APILocator.getUserAPI().getSystemUser(), false);
+
+                final Container container = APILocator.getContainerAPI()
+                        .getWorkingContainerById(tree.getContainer(),
+                                APILocator.getUserAPI().getSystemUser(), false);
+
+                if (InodeUtils.isSet(page.getInode()) && InodeUtils.isSet(container.getInode())) {
```

Comment:
it is a very small thing but seems it is a shortcircuit it would be worthy to try this in two if such as 

```
final IHTMLPage page = APILocator.getHTMLPageAssetAPI()
                        .findByIdLanguageFallback(tree.getHtmlPage(), contentlet.getLanguageId(),
                                false,
                                APILocator.getUserAPI().getSystemUser(), false);

if (InodeUtils.isSet(page.getInode()) {
    final Container container = APILocator.getContainerAPI()
                        .getWorkingContainerById(tree.getContainer(),
                                APILocator.getUserAPI().getSystemUser(), false);
    if (InodeUtils.isSet(container.getInode())) {
        ......

```

Minor thing, but if the page inode is not set, you will save CPU time not asking for the container

Reply:
This is not new code @jdotcms , but makes sense to me. I will apply it

Reply:
Done